File: fixtures/fortran/file1.f
      PROGRAM MATHUTILS
C     Common math utility subroutines
      IMPLICIT NONE

      SUBROUTINE COMPUTE_STATS(ARR, N, MEAN, STDDEV)
      INTEGER N
      REAL ARR(N), MEAN, STDDEV
      REAL SUM, SUMSQ
      INTEGER I
      SUM = 0.0
      SUMSQ = 0.0
      DO I = 1, N
          SUM = SUM + ARR(I)
          SUMSQ = SUMSQ + ARR(I) * ARR(I)
      END DO
      MEAN = SUM / N
      STDDEV = SQRT(SUMSQ / N - MEAN * MEAN)
      RETURN
      END

      SUBROUTINE SORT_ARRAY(ARR, N)
      INTEGER N
      REAL ARR(N), TEMP
      INTEGER I, J
      DO I = 1, N - 1
          DO J = 1, N - I
              IF (ARR(J) .GT. ARR(J+1)) THEN
                  TEMP = ARR(J)
                  ARR(J) = ARR(J+1)
                  ARR(J+1) = TEMP
              END IF
          END DO
      END DO
      RETURN
      END

C     File-specific: print array
      SUBROUTINE PRINT_ARRAY(ARR, N)
      INTEGER N, I
      REAL ARR(N)
      DO I = 1, N
          WRITE(*,*) 'ARR(', I, ') =', ARR(I)
      END DO
      RETURN
      END
      END PROGRAM MATHUTILS

File: fixtures/pascal/file1.pas
program DataProcessor;

uses
  SysUtils, Classes, Math;

type
  TDataRecord = record
    Id    : Integer;
    Name  : string;
    Value : Double;
    Active: Boolean;
  end;

function ValidateInput(const Value: string): Boolean;
begin
  Result := (Length(Value) > 0) and (Length(Value) <= 255);
end;

function ParseIntSafe(const S: string; out Value: Integer): Boolean;
begin
  Result := TryStrToInt(S, Value);
end;

procedure SortByValue(var Arr: array of Double);
var
  I, J  : Integer;
  Temp  : Double;
begin
  for I := Low(Arr) to High(Arr) - 1 do
    for J := Low(Arr) to High(Arr) - I - 1 do
      if Arr[J] > Arr[J + 1] then
      begin
        Temp     := Arr[J];
        Arr[J]   := Arr[J + 1];
        Arr[J+1] := Temp;
      end;
end;

// File-specific: compute average
function ComputeAverage(const Arr: array of Double): Double;
var
  I   : Integer;
  Sum : Double;
begin
  Sum := 0;
  for I := Low(Arr) to High(Arr) do
    Sum := Sum + Arr[I];
  Result := Sum / Length(Arr);
end;

begin
  WriteLn('Data Processor v1');
end.
